### ISSUE TITLE
groovy mocks are not truthy

### ISSUE THREAD

**Issue by robfletcher** [spockframework/spock#438]
Originally reported on Google Code with ID 316

```
See the example below.
Regular mocks evaluate to true but groovy mocks resolve to false.

class GroovyMockSpec extends Specification {
    def mock = Mock(A)
    def groovyMock = GroovyMock(Object)

    def "regular mocks are truthy"() {
        expect:
        mock ?: 'irrelevant' == mock
    }

    def "groovy mocks are falsey"() {
        expect:
        groovyMock ?: 'unexpected' == 'unexpected'
    }

    static interface A {}
}

What version of Spock and Groovy are you using?
0.7-groovy-1.8
```

Reported by `sgo@thinkerit.be` on 2013-06-17 12:36:31

**Comment by steshaw** [comment]
Should this issue still be open?

**Comment by Vampire** [comment]
Well, the behavior at least is still the same.
It's just the question whether this should be changed or not. 

I personally tend to say the behavior is fine, that's just the difference between Java and Groovy objects. The Java object does not respond to `asBoolean` so truthy means non-null, the Groovy object responds to `asBoolean`, so it is used and as mock object answers with `false` by default.

**Comment by steshaw** [comment]
Yes, 10 years is a long time to live with the behaviour if it's so bad. I imagine that it violates the principle of least surprise, though. Would there be a way to fix it? i.e. is it possible to detect Java objects and use a Groovy wrapper or something to respond in the same way as a Groovy object?

**Comment by Vampire** [comment]
Any behavior different of the current one would for me violate the principle of least surprise to be honest.

**Comment by steshaw** [comment]
Yes, at this point, I suppose it would. This leads me to the conclusion that this issue is a will-not-fix.
